opm/simulators/flow/LogOutputHelper.cpp

```diff
@@ -436,15 +465,36 @@ injection(const std::size_t reportStepNum) const
 
         // values[Ix::SteadyStateII] = 0;
 
-        this->outputInjectionReportRecord_(values, names);
+        std::vector<ConnData> connData;
+        if (withConns) {
+            const auto& connections = well.getCompletions();
```

Comment:
If you don't really care about the "lumped" completions (one or more connections grouped into a larger unit), then you can use [`Well::getConnections()`](https://github.com/OPM/opm-common/blob/895d2bd1883263a86eb80d57646a14d86ab68e5f/opm/input/eclipse/Schedule/Well/Well.hpp#L465) instead of `Well:getCompletions()`.  That, moreover, has the added benefit of not creating copies of the `Connection` objects.

Using `Well::getConnections()` you can write this loop (a little) simpler as
```C++
const auto& units = ...;
for (const auto& connection : well.getConnections()) {
    auto& conn = connData.emplace_back(connection);
    // ... populate 'conn' ...
}
```